vars/pushDockerImages.txt

```diff
@@ -0,0 +1,38 @@
+Publish docker images in the given docker registry. For such, it
+retags the existing docker images and publish them in the given
+docker namespace.
+
+It also allows to transform the version and customise private docker
+namespaces/registries.
+
+```
+  // Given the filebeat project, and its generated docker
+  // images for the 8.2.0-SNAPSHOT and 4 different variants
+  // then publish them to observability-ci
+  pushDockerImages(
+    secret: "my-secret",
+    registry: "my-registry",
+    arch: 'amd64',
+    version: '8.2.0',
+    snapshot: true,
+    imageName: 'filebeat',
+    variants: [
+      '' : 'beats',
+      'ubi8' : 'beats',
+      'cloud' : 'beats-ci',
+      'complete' : 'beats',
+    ],
+    targetNamespace: 'observability-ci'
+  )
+```
+
+* secret: the docker secret
+* registry: the docker registry
+* arch: the supported arch (amd64 or arm64)
+* version: what version
+* snapshot: snapshot support
+* imageName: the docker image name.
+* variants: variants and docker namespace to copy from.
+* targetNamespace: what docker namespace to publish to.
```

Comment:
`arch` and `variant` are complicated, it is not obvious what will be the tag, maybe it worth a few examples to understand how it works

Reply:
The above example is not enough?

Reply:
I do not think so, you do not know which Docker image is the `source` and which will be the `destination` if you do not check the code.